.circleci/config.yml

```diff
@@ -1,111 +1,96 @@
 
+parameters:
+  boost-libs:
+    description: "List of boost libraries to build"
+    type: string
+    # We removed serialization as we need to be able to build for iOS 9 and there are code that is using iOS 10+ features
+    default: "atomic context coroutine date_time exception iostreams program_options random regex system test thread"
+
+  xcode-version:
+    description: Xcode version for iOS and Darwin builds
+    type: string
+    default: "12.5.1"
+
+  twilio-video-sdk-toolchain-image:
+    description: twilio-video-sdk-toolchain docker image to use for linux and android
+    type: string
+    default: "twilio/twilio-video-sdk-toolchain:17"
+
 aliases:
   - &workspace
     ~/twilio-boost-build
-  - &build_output
-    ~/twilio-boost-build/target/output
+
   - ignore-master: &ignore-master
       filters:
         tags:
           only: /.*/
         branches:
           ignore: /^master/
+
   - only-release-tags: &only-release-tags
       filters:
         tags:
           only: /^release-.*/
         branches:
           ignore: /.*/
-  - platform-parameter: &platform-parameter
-      platform:
-        description: "Platform to build"
-        type: enum
-        enum: ["headers", "android", "linux", "linux-cxx11-abi-disabled", "ios", "osx"]
-  - boost-libs-parameter: &boost-libs-parameter
-      boost-libs:
-        description: "List of boost libraries to build"
-        type: string
-        # We removed serialization as we need to be able to build for iOS 9 and there are code that is using iOS 10+ features
-        default: "atomic context coroutine date_time exception iostreams program_options random regex system test thread"
-  - pre-steps-parameter: &pre-steps-parameter
-      pre-steps:
-        description: "Steps that will be executed before build starts"
-        type: steps
-        default: []
-  - post-steps-parameter: &post-steps-parameter
-      post-steps:
-        description: "Steps that will be executed after build ends"
-        type: steps
-        default: []
-  - build-steps-parameter: &build-steps-parameter
-      build-steps:
-        description: "Steps that will be executed as main build target"
-        type: steps
 
 executors:
   linux-executor:
+    parameters:
+      resource-class:
+        type: enum
+        enum: [small, medium, medium+, large, xlarge]
+        default: medium+
     docker:
-      - image: cibuilderbot/docker-circleci-linux-android
-        auth:
-          username: $DOCKER_HUB_USERNAME
-          password: $DOCKER_HUB_PASSWORD
-    resource_class: xlarge
-    working_directory: *workspace
-    environment:
-      CMAKE_BUILD_PARALLEL_LEVEL: 2
-  android-executor:
-    docker:
-      - image: twilio/twilio-video-sdk-toolchain:16
+      - image: << pipeline.parameters.twilio-video-sdk-toolchain-image >>
         auth:
           username: $DOCKER_HUB_USERNAME
           password: $DOCKER_HUB_PASSWORD
-    resource_class: xlarge
+    resource_class: << parameters.resource-class >>
```

Comment:
What is the benefit in keeping it configurable? The default medium+ isn't good for all cases?

Reply:
The `unpack-sources` job needs only a `small` instance.

Reply:
ah. got it.